contract/src/events/events_cep47.rs

```diff
@@ -0,0 +1,122 @@
+use alloc::{
+    collections::BTreeMap,
+    format,
+    string::{String, ToString},
+};
+use casper_contract::{
+    contract_api::{
+        runtime,
+        storage::{dictionary_get, dictionary_put, new_dictionary},
+    },
+    unwrap_or_revert::UnwrapOrRevert,
+};
+use casper_types::Key;
+
+use crate::{
+    constants::{CEP78_PREFIX, HASH_KEY_NAME_1_0_0},
+    error::NFTCoreError,
+    modalities::TokenIdentifier,
+    utils::{self, get_stored_value_with_user_errors},
+};
+
+pub(crate) enum CEP47Event {
+    Mint {
+        recipient: Key,
+        token_id: TokenIdentifier,
+    },
+    Burn {
+        owner: Key,
+        token_id: TokenIdentifier,
+    },
+    Approve {
+        owner: Key,
+        spender: Key,
+        token_id: TokenIdentifier,
+    },
+    Transfer {
+        sender: Key,
+        recipient: Key,
+        token_id: TokenIdentifier,
+    },
+    MetadataUpdate {
+        token_id: TokenIdentifier,
+    },
+}
+
+pub(crate) fn record_event_dictionary(event: &CEP47Event) {
+    let collection_name: String = get_stored_value_with_user_errors(
+        crate::constants::COLLECTION_NAME,
+        NFTCoreError::MissingCollectionName,
+        NFTCoreError::InvalidCollectionName,
+    );
+
+    let package = utils::get_stored_value_with_user_errors::<String>(
+        &format!("{}{}", CEP78_PREFIX, collection_name),
+        NFTCoreError::MissingCep78PackageHash,
+        NFTCoreError::InvalidCep78InvalidHash,
+    );
+
+    let event: BTreeMap<&str, String> = match event {
+        CEP47Event::Mint {
+            recipient,
+            token_id,
+        } => {
+            let mut event = BTreeMap::new();
+            event.insert(HASH_KEY_NAME_1_0_0, package);
```

Comment:
Lets replace this with `cep78_contract_package` instead of using the legacy scheme